### DISCUSSION TITLE
Concerns raised in Salaboy's article, and a reasoned response

### DISCUSSION THREAD

**Discussion by squeedee** [vmware-tanzu/cartographer#844] [General]
In [Salaboy's Knative Weekly discussion about CD](https://salaboy.com/2022/05/07/knative-oss-weekly-38/)  he raises some valid points of difference between Cartographer and other CD tools, especially Keptn. 

I'ld like to gather a reasoned response to what he has to say here, and perhaps finish up by posting something in response to his article.

**Comment by squeedee** [comment]
To capture what I said in the Carto Office Hours.

Cartographer's supply-chains (blueprints) aim to separate the "Definition of a path to production" from the mundane "breaking ecosystem boundaries to get to the right tools".  Currently our catalog of useful templates is quite small (and mostly captured in examples or the [TCE Catalog](https://github.com/vmware-tanzu/cartographer-catalog)(public soon)) - With a growth of kubernete's native controllers for supply chain tasks, and gap-bridging tasks using Tekton or jobs, the apparent value of this benefit should grow.

**Comment by davidmirror-ops** [comment]
Also, from [Office Hours](https://hackmd.io/g5EjHok9SiKqlwV846QlvA?view) notes:

The surface of both Keptn and Carto are events, the difference is that Carto is [level-triggered](https://hackernoon.com/level-triggering-and-reconciliation-in-kubernetes-1f17fe30333d) vs having to capture events to get status (edge-triggered). Most of the RFCs authored recently aim to simplify integrating external tooling, not necessarily K8s native.

RFCs:
1. Simplify Runnable usage: https://github.com/vmware-tanzu/cartographer/pull/705
2. Add Template field to Runnable: https://github.com/vmware-tanzu/cartographer/pull/787
